Code/GraphMol/Fingerprints/Fingerprints.cpp

```diff
@@ -173,6 +173,182 @@ return res;
 
 }  // end of anonymous namespace
 
+
+namespace utils {
+
+void buildDefaultAtomInvariants(const ROMol &mol,
+  std::vector<boost::uint32_t>& lAtomInvariants){
+
+  lAtomInvariants.reserve(mol.getNumAtoms());
```

Comment:
In case this function ever ends up in a public API, you really should clear `lAtomInvariants` before you start pushing stuff onto it.